x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -829,7 +829,12 @@ private static String sslContextAlgorithm(List<String> supportedProtocols) {
     }
 
     private boolean shouldEnableDiagnoseTrust() {
-        if (XPackSettings.FIPS_MODE_ENABLED.get(settings) && DIAGNOSE_TRUST_EXCEPTIONS_SETTING.exists(settings) == false ) {
+        // We disable the DiagnosticTrustManager in tests in Java 8 in FIPS 140 mode, as we're not allowed to wrap X509TrustManager
+        final boolean explicitlyDisable = Boolean.parseBoolean(System.getProperty("es.disable.diagnostic.trust.manager", "false"));
```

Comment:
Why can't we pass in fips mode as disabled in our tests properties from BuildPlugin? I don't see the need for another property, and don't like this leaking into production code.

Reply:
Because we need this to be set in all of our tests even when we ie build an SslService from an arbitrary settings object, and doing so in build plugin isn't enough, the previous two PRs contain much more detail, let me know if I should elaborate.

I didn't reuse the tests.fips.enabled setting do that this can be selectively set when running in java 8 only. Leaking into production code wasnt my first option either but the alternatives were brittle for the reasons I explain in the description of the previous PR

Reply:
If java 8 is the problem, shouldn't we always disable the diagnostic trust manager for that version? Why does this only apply to tests?

Reply:
Its java 8 with with SunJSSE in fips mode that is problematic, I explain this in detail in the previous two PRs, in retrospect I should have carried that discussion over to this PR. 

We don't want to arbitrarily assume that there is no other way to run in fips mode in java 8, even if the problematic one is one of the most popular, so we didn't want to disable the diagnostic trust manager for all cases. 

Maybe a short sync chat is worth to give you the whole context if you still have concerns about this ?